### PR TITLE
Release 0.4.0

### PR DESCRIPTION
**Pull request by jlowe**
This merges branch-0.4 to the main branch and updates the project version to 0.4.0.

**NOTE: This should be committed as a merge commit, not squashed!**

### CODE REVIEW

**Comment by jlowe** [comment]
build

**Review by revans2 — APPROVED** [review]

File: shims/spark300/pom.xml [hunk 1]
```diff
@@ -22,14 +22,14 @@
     <parent>
         <groupId>com.nvidia</groupId>
```
Comment (tgravescs) [inline shims/spark300/pom.xml:23]: update copyright to 2021

File: shims/spark300emr/pom.xml [hunk 2]
```diff
@@ -22,14 +22,14 @@
     <parent>
         <groupId>com.nvidia</groupId>
         <artifactId>rapids-4-spark-shims_2.12</artifactId>
-        <version>0.3.0</version>
```
Comment (tgravescs) [inline shims/spark300emr/pom.xml:25]: copyright

File: shims/spark301/pom.xml [hunk 3]
```diff
@@ -22,14 +22,14 @@
     <parent>
         <groupId>com.nvidia</groupId>
         <artifactId>rapids-4-spark-shims_2.12</artifactId>
-        <version>0.3.0</version>
```
Comment (tgravescs) [inline shims/spark301/pom.xml:25]: copyright needs update

File: shims/spark301emr/pom.xml [hunk 4]
```diff
@@ -22,14 +22,14 @@
     <parent>
         <groupId>com.nvidia</groupId>
         <artifactId>rapids-4-spark-shims_2.12</artifactId>
-        <version>0.3.0</version>
+        <version>0.4.0</version>
```
Comment (tgravescs) [inline shims/spark301emr/pom.xml:25]: copyright

File: shims/spark302/pom.xml [hunk 5]
```diff
@@ -22,14 +22,14 @@
     <parent>
         <groupId>com.nvidia</groupId>
         <artifactId>rapids-4-spark-shims_2.12</artifactId>
-        <version>0.3.0</version>
```
Comment (tgravescs) [inline shims/spark302/pom.xml:25]: copyright

File: shuffle-plugin/pom.xml [hunk 6]
```diff
@@ -22,14 +22,14 @@
     <parent>
         <groupId>com.nvidia</groupId>
         <artifactId>rapids-4-spark-parent</artifactId>
-        <version>0.3.0</version>
```
Comment (tgravescs) [inline shuffle-plugin/pom.xml:25]: copyright

File: sql-plugin/pom.xml [hunk 7]
```diff
@@ -22,13 +22,13 @@
     <parent>
         <groupId>com.nvidia</groupId>
         <artifactId>rapids-4-spark-parent</artifactId>
-        <version>0.3.0</version>
```
Comment (tgravescs) [inline sql-plugin/pom.xml:25]: copyright

File: udf-compiler/pom.xml [hunk 8]
```diff
@@ -22,13 +22,13 @@
     <parent>
         <groupId>com.nvidia</groupId>
         <artifactId>rapids-4-spark-parent</artifactId>
-        <version>0.3.0</version>
```
Comment (tgravescs) [inline udf-compiler/pom.xml:25]: copyright

**Comment by jlowe** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Review by sameerz — APPROVED** [review]

**Pull request merged**